docs/docsite/rst/dev_guide/developing_modules_documenting.rst

```diff
@@ -10,6 +10,88 @@ which conforms to the schema defined below. You may find it easier to
 start writing your ``DOCUMENTATION`` string in an editor with YAML
 syntax highlighting before you include it in your Python file.
 
+All modules must have the following sections defined in this order:
+
+1. ANSIBLE_METADATA
+2. DOCUMENTATION
+3. EXAMPLES
+4. RETURNS
+5. Python imports
+
+.. note:: Why don't the imports go first?
+
+  Keen Python programmers may notice that contary to PEP8's advice we don't put ``imports`` at the top of the file. This is due to how modules used to be combined with libraries. For more information see :doc:`developing_program_flow_modules`.
+
```

Comment:
Just to expand on this, some excerpts from my proposal:

> We have decided to continue to ignore E402 module level import not at top of file due to our preference to place imports in modules below the DOCUMENTATION/RETURN/EXAMPLES/METADATA variables, as we treat those similarly to module level docstrings. This allows us to keep all code and imports together instead of split from each other.

> The decision to exclude E402 is based on readability. As described above, documentation strings in a module are much more similar to module level docstrings, than code, and are never utilized by the module itself. Placing the imports below this documentation and closer to the code, consolidates and groups all related code in a congruent manner to improve readability, debugging and understanding.